controllers/idler/idler_controller.go

```diff
@@ -141,6 +155,64 @@ func (r *Reconciler) ensureIdling(logger logr.Logger, idler *toolchainv1alpha1.I
 	return r.updateStatusPods(idler, newStatusPods)
 }
 
+func (r *Reconciler) createNotification(logger logr.Logger, idler *toolchainv1alpha1.Idler) error {
+	//Get the HostClient
+	hostCluster, ok := r.GetHostCluster()
```

Comment:
wouldn't it be simpler if `r.GetHostCluster()` returns `hostCluster, err`?
In other words, in line 47 above:
```
GetHostCluster      func() (*cluster.CachedToolchainCluster, error)
```

Reply:
yes, indeed